lib/client/client.go

```diff
@@ -254,12 +268,18 @@ func (proxy *ProxyClient) reissueUserCerts(ctx context.Context, params ReissuePa
 	return key, nil
 }
 
+type PromptMFAChallengeHandler func(ctx context.Context, proxyAddr string, c *proto.MFAAuthenticateChallenge, promptDevicePrefix string) (*proto.MFAAuthenticateResponse, error)
+
 // IssueUserCertsWithMFA generates a single-use certificate for the user.
-func (proxy *ProxyClient) IssueUserCertsWithMFA(ctx context.Context, params ReissueParams) (*Key, error) {
-	localAgent := proxy.teleportClient.LocalAgent()
-	key, err := localAgent.GetKey(WithKubeCerts(params.RouteToCluster))
-	if err != nil {
-		return nil, trace.Wrap(err)
+func (proxy *ProxyClient) IssueUserCertsWithMFA(ctx context.Context, params ReissueParams, promptMFAChallenge PromptMFAChallengeHandler) (*Key, error) {
+	key := params.ExistingCreds
+	if key == nil {
+		localAgent := proxy.teleportClient.LocalAgent()
+		var err error
+		key, err = localAgent.GetKey(WithKubeCerts(params.RouteToCluster))
```

Comment:
According to the code below, it's possible that `params.RouteToCluster` will be empty here. Is that gonna be handled properly?

Reply:
Added defaulting logic for `params.RouteToCluster`